src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -95,13 +105,43 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $kernelDefinition = $container->getDefinition('kernel');
             $kernelDefinition->addTag('routing.route_loader');
 
-            if ($this instanceof EventSubscriberInterface) {
-                $kernelDefinition->addTag('kernel.event_subscriber');
+            $container->addObjectResource($this);
+            $container->fileExists($this->getProjectDir().'/config/bundles.php');
+            $container->setParameter('container.dumper.inline_class_loader', \PHP_VERSION_ID < 70400 || !ini_get('opcache.preload'));
+            $container->setParameter('container.dumper.inline_factories', true);
+            $container->setParameter('kernel.secret', '%env(APP_SECRET)%');
+
+            try {
+                $this->configureContainer($container, $loader);
+
+                return;
+            } catch (\TypeError $e) {
+                $file = $e->getFile();
+
+                if (0 !== strpos($e->getMessage(), sprintf('Argument 1 passed to %s::configureContainer() must be an instance of %s,', static::class, ContainerConfigurator::class))) {
+                    throw $e;
+                }
             }
 
-            $this->configureContainer($container, $loader);
+            $defaultDefinition = (new Definition())->setAutowired(true)->setAutoconfigured(true);
+            $kernelLoader = $loader->getResolver()->resolve($file);
```

Comment:
```
/* @var PhpFileLoader $kernelLoader */
```

it makes the logic a bit easier to follow... 😇

Reply:
added